### PR TITLE
CONTRIB: add --enable-examples to configure-devel

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## Why ?
Useful to have built examples during development, for example, to check compatibility.

@amaslenn @yosefe

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8516/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8517/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8524/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8532/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
@yosefe pls review coverity related changes

File: test/examples/ucx_hello_world.h [hunk 1]
```diff
@@ -94,11 +94,18 @@ int client_connect(const char *server, uint16_t server_port)
...
-    recv(oob_sock, &dummy, sizeof(dummy), 0);
+    ssize_t res;
+
+    res = send(oob_sock, &dummy, sizeof(dummy), 0);
+    if (res < 0) {
+        return res;
+    }
+
+    res = recv(oob_sock, &dummy, sizeof(dummy), 0);
+    return !(res == sizeof(dummy));
```
Comment (yosefe) [inline test/examples/ucx_hello_world.h]: pls add comment

File: test/examples/ucx_hello_world.h [hunk 2]
```diff
@@ -112,6 +119,7 @@ static void generate_random_string(char *str, int size)
     }
 
     for (i = 0; i < (size-1); ++i) {
+        /* coverity[dont_call] */
```
Comment (yosefe) [inline test/examples/ucx_hello_world.h]: can we avoid coverity suppression and fix in another way?

File: test/examples/ucx_hello_world.h (hunk 1 above)
Comment (evgeny-leksikov) [reply test/examples/ucx_hello_world.h]: done

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8550/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8555/ for details (Mellanox internal link).

**Pull request merged**